spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/DataFileServices.java

```diff
@@ -0,0 +1,132 @@
+
+    // Vectorized readers
+    DataFileServiceRegistry.registerReader(
+        FileFormat.PARQUET,
+        ColumnarBatch.class.getName(),
+        ParquetReaderType.ICEBERG.name(),
+        inputFile ->
+            Parquet.read(inputFile)
+                .batchReaderFunction(
+                    (schema, messageType, idToConstant, deleteFilter) ->
+                        VectorizedSparkParquetReaders.buildReader(
+                            schema,
+                            messageType,
+                            idToConstant,
+                            (DeleteFilter<InternalRow>) deleteFilter)));
+
+    DataFileServiceRegistry.registerReader(
+        FileFormat.PARQUET,
+        ColumnarBatch.class.getName(),
+        ParquetReaderType.COMET.name(),
+        inputFile ->
+            Parquet.read(inputFile)
+                .batchReaderFunction(
+                    (schema, messageType, idToConstant, deleteFilter) ->
+                        VectorizedSparkParquetReaders.buildCometReader(
+                            schema,
+                            messageType,
+                            idToConstant,
+                            (DeleteFilter<InternalRow>) deleteFilter)));
+
+    DataFileServiceRegistry.registerReader(
+        FileFormat.ORC,
+        ColumnarBatch.class.getName(),
+        inputFile ->
+            ORC.read(inputFile).batchReaderFunction(VectorizedSparkOrcReaders::buildReader));
+
+    DataFileServiceRegistry.registerAppender(
+        FileFormat.AVRO,
+        InternalRow.class.getName(),
+        outputFile ->
+            Avro.write(outputFile)
+                .writerFunction(
+                    (schema, engineSchema) -> new SparkAvroWriter((StructType) engineSchema))
+                .deleteRowWriterFunction(
+                    (schema, engineSchema) ->
+                        new SparkAvroWriter(
+                            (StructType)
+                                ((StructType) engineSchema)
+                                    .apply(DELETE_FILE_ROW_FIELD_NAME)
+                                    .dataType())));
+
+    DataFileServiceRegistry.registerAppender(
+        FileFormat.PARQUET,
+        InternalRow.class.getName(),
+        outputFile ->
+            Parquet.write(outputFile)
+                .writerFunction(
+                    (engineSchema, icebergSchema, messageType) ->
+                        SparkParquetWriters.buildWriter((StructType) engineSchema, messageType))
+                .pathTransformFunc(path -> UTF8String.fromString(path.toString())));
+
+    DataFileServiceRegistry.registerAppender(
+        FileFormat.ORC,
+        InternalRow.class.getName(),
+        outputFile ->
+            ORC.write(outputFile)
+                .writerFunction(
+                    (schema, messageType, engineSchema) -> new SparkOrcWriter(schema, messageType))
+                .pathTransformFunc(path -> UTF8String.fromString(path.toString())));
```

Comment:
What is this?

Reply:
The delete writer needs this to convert the path to strings in the delete records.
All of these are file format specific, so I think this should be fine here as this is not exposed to the users